edit mode: option for additional folder (font-search path)

artandtechvof opened the pull request:
Made addon folder selection option for sound fonts, for common menu sounds etc.
 Changed order of edit-menu items to better fit the new font + addon selection box

profezzorn commented on app.html:
```diff
@@ -558,16 +560,37 @@
-    Send("set_font " + font_select.value);
+  var addon_select = document.getElementById('font_select2');
+  var font_path = font_select.value;
+  if (addon_select.value != "none"){
```
space between ) and {

profezzorn commented on app.html:
```diff
@@ -558,16 +560,37 @@
+  var tmp = value.split(";");
+  var font_ = value;
+  var addon_ = "none"
+  if (tmp.length > 1){
```
space between ) and {

profezzorn commented on app.html:
```diff
@@ -558,16 +560,37 @@
+  var addon_ = "none"
+  if (tmp.length > 1){
+    for (var i = 0; i < font_lines.length; i++) {
+      if (tmp[0] == font_lines[i]) { font_ = tmp[0];value = tmp[1];}
```
space between ; and "value" (same on next line)

profezzorn commented on app.html:
```diff
@@ -558,16 +560,37 @@
+  var tmp = value.split(";");
+  var font_ = value;
+  var addon_ = "none"
+  if (tmp.length > 1){
```
What happens if there are more than 2 components in the font search path?

artandtechvof replied:
You got me there. I was still debating how to solve that one. I could imagine an "+" sign in the edit box to add additional folders. After clicking the "+" the pull down menu will show  to select the additional folder. The "+" will shift down allowing you to add as many search folder as you want. Nested folders within other soundfonts would be challenging, unless we allow for a text-input edit field to specify the folders you want to include (thsi is mor prone to erroneous entries)
Doing so would also solve the necessity to have a "none" showing currently in the additional folder if you don't have a common folder for any of your sounds. 
For now most likely is that you have your menu sounds in a common folder. With the previous version of WebAPP you would loose you common folder, by the way the pulldown only selects a single sound font.

Something for the ToDo list...

profezzorn replied:
I think it's fine to only have two selectboxes, but it should always be possible to select what was already there, and perhaps also what was listed in other presets.  So like if the path is foo;bar;gazonk, maybe the top selectbox would say "foo" and the bottom one would say "bar;gazonk" ?  Or, we could just have "foo;bar;gazonk" in the top box and None in the bottom. Either way, I don't think it's acceptable for the "gazonk" to just disappear.

artandtechvof replied:
In the current version the check box indeed shows the whole path (although corrupted because of the starjedi font not showing ";" correctly. As soon as you alter the font, it will overwrite with the full path, with only the selected font.
Having the entries of the other presets selectable would be a nice idea to keep the full path with multiple search folders. 
Is there a way to get the star-jedi to show ";" correctly?

profezzorn replied:
I don't think StarJedi has ; unfortunately.
Should be possible to switch in a font just for that character though.
I'm ok with switching fonts changing the number of font elements, as long as you can switch back too.

profezzorn commented on app.html:
```diff
@@ -558,16 +560,37 @@
+  var font_ = value;
+  var addon_ = "none"
+  if (tmp.length > 1){
+    for (var i = 0; i < font_lines.length; i++) {
```
Could you add some comments to explain what these are for.
I think I get it, but it's probably better to explain your intentions than having to infer it from the code.

artandtechvof replied:
done

artandtechvof commented:
font search path broken down into multiple sections, order as found in the preset
font folders can be added by '+' button
presets that have font with search-folders are listed to top of listed
modified the font for select_list items, for better readability in combination with the ';'

I think this is a nice way of having the items we discussed earlier. You can directly select any of the font-searchpath from other presets, build your own, and have a variable number of search folders.

artandtechvof commented:
Please have a look at this latest version

modified font in selection field, for better readability when a ";" is present

modifed font selection fields, multiple folder fonts are now supportted
selecting a predifined multiple folder font is supoorted
adding or removing search folders is supported

changed order of edit items for better readability with larger multiple folder fonts

You can testdrive it at;

https://artandtechvof.github.io/app_fontsearch.html

NoSloppy commented:
> You can testdrive it at;
> 
> https://artandtechvof.github.io/app_fontsearch.html

FYI, all effects except clash can be played while the saber is OFF, and stab causes ignition and only a second press plays stab sound.
